src/CharityVault.sol

```diff
@@ -143,96 +185,166 @@ contract CharityVault is ERC20, Auth {
-        // Multiply by 1 - the percent to be donated to charity
-        return (getFloat() + vault.totalDeposited() - vault.calculateLockedProfit()) * ((100.0 - feePercent) / 100.0);
+        // We subtract locked profit from totalStrategyHoldings because maxLockedProfit is baked into it.
+        return totalFloat() + (totalStrategyHoldings - lockedProfit());
     }
 
-    /// @notice Calculate the total amount of free underlying tokens.
-    function calculateCharityTotalFreeUnderlying() public view returns (uint256) {
-        // Subtract locked profit from the amount of total deposited tokens and add the float value.
-        // We subtract the locked profit from the total deposited tokens because it is included in totalDeposited.
-        // Multiply by the percent to be donated to charity
-        return (getFloat() + vault.totalDeposited() - vault.calculateLockedProfit()) * (feePercent / 100.0);
+    /// @notice Calculate the current amount of locked profit.
+    /// @return The current amount of locked profit.
+    function lockedProfit() public view returns (uint256) {
+        // TODO: Cache SLOADs?
+        return
+            block.timestamp >= lastHarvest + profitUnlockDelay
+                ? 0 // If profit unlock delay has passed, there is no locked profit.
+                : maxLockedProfit - (maxLockedProfit * (block.timestamp - lastHarvest)) / profitUnlockDelay;
     }
 
+    /// @notice Returns the amount of underlying tokens that idly sit in the Vault.
+    /// @return The amount of underlying tokens that sit idly in the Vault.
+    function totalFloat() public view returns (uint256) {
+        return UNDERLYING.balanceOf(address(this));
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                        TRANSPARENT FALLBACK FUNCTIONALITY
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Erroneous ether sent will be forward to the charity as a donation
+    receive() external payable {
+        (bool sent, bytes memory _data) = CHARITY.call{value: msg.value}("");
+        require(sent, "Failed to send to CHARITY");
 
+        // If sent, emit logging event
+        emit TransparentTransfer(msg.sender, msg.value);
+    }
 
-    receive() external payable {}
+    /// @notice Forwards any unknown calls to the underlying VAULT
+    /// @dev Uses the Solidity 0.6 receive split fallback functionality as specified in the blog
+    /// @dev https://blog.soliditylang.org/2020/03/26/fallback-receive-split/
+    fallback() external payable {
+            assembly {
+                calldatacopy(0, 0, calldatasize())
+                let result := delegatecall(gas(), payable(address(VAULT)), 0, calldatasize(), 0, 0)
+                returndatacopy(0, 0, returndatasize())
+                switch result
+                case 0 { revert(0, returndatasize()) }
+                default { return(0, returndatasize()) }
+            }
+        }
 }
```

Comment:
lol interesting but why would people want this? feels unnecessarily risky

Reply:
haha just me experimenting

Reply:
will remove